data/nat/choose.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+Mostly based on Jeremy Avigad's choose file in lean 2
+-/
+
+
+import data.nat.basic
+
+open nat
+def choose : ℕ → ℕ → ℕ
+| _              0  := 1
+| 0        (succ k) := 0
+| (succ n) (succ k) := choose n k + choose n (succ k)
+
+@[simp]
+lemma choose_zero_right (n : ℕ) : choose n 0 = 1 := begin
+  cases n with n,unfold choose,unfold choose,
+end
+
+@[simp]
+lemma choose_zero_succ (k : ℕ) : choose 0 (succ k) = 0 := rfl
+
+lemma choose_succ_succ (n k : ℕ) : choose (succ n) (succ k) = choose n k + choose n (succ k) := rfl
+
+lemma choose_eq_zero_of_lt {n k : ℕ} : n < k → choose n k = 0 := begin
+  revert k,
+  induction n with n' hi,
+  assume k nk,cases k,exact absurd nk dec_trivial,
```

Comment:
avoid writing multiple tactics in one line.
Here you use multiple tactics to proof the first induction step. Please mark it with `{ ... }`  and indent it properly.

Reply:
Also, space after `,`

Reply:
This also goes for further proofs.